url.bs

```diff
@@ -2298,6 +2298,10 @@ these steps:
 
        <li><p>If <a>c</a> is not U+002F (/), then decrease <var>pointer</var> by 1.
       </ol>
+
+     <li><p>Otherwise, if <var>state override</var> is given and <var>url</var>'s
+      <a for=url>host</a> is null, <a for=list>append</a> the empty string to <var>url</var>'s
+      <a for=url>path</a>
```

Comment:
Why only when host is null? It seems this needs to generally happen.

Reply:
Because non-special URLs with a host (e.g. `foo://somehost`) can have an empty string as a path. The issue is limited to path-only URLs.